src/BloomBrowserUI/bookEdit/js/textOverPicture.ts

```diff
@@ -31,16 +40,237 @@ class TextOverPictureManager {
+        const focusedElement = event.currentTarget as Element; // The current target is the element we attached the event listener to
+        if (focusedElement.classList.contains("bloom-editable")) {
+            // If we focus something on the page that isn't in a bubble, we need to switch
+            // to having no active bubble element. Note: we don't want to use focusout
+            // on the bubble elements, because then we lose the active element while clicking
+            // on controls in the toolbox (and while debugging).
+            // Todo: check what happens when a bubble is removed.
+            initializeTextOverPictureManager();
+
+            const bubbleElement = focusedElement.closest(
+                ".bloom-textOverPicture"
+            );
+            if (bubbleElement) {
+                theOneTextOverPictureManager.setActiveElement(
+                    bubbleElement as HTMLElement
+                );
+            } else {
+                theOneTextOverPictureManager.setActiveElement(undefined);
+            }
+        }
+    }
+
+    public getActiveElement() {
+        return this.activeElement;
+    }
+
+    private setActiveElement(element: HTMLElement | undefined) {
+        if (element == undefined) {
+            this.activeElement = undefined;
+            return;
+        }
+
+        this.activeElement = element;
+        if (this.notifyBubbleChange) {
+            this.notifyBubbleChange(this.getSelectedItemBubbleSpec());
+        }
+    }
+
+    public turnOffBubbleEditing(): void {
+        if (this.isCalloutEditingOn === false) {
+            return; // Already off. No work needs to be done.
+        }
+        this.isCalloutEditingOn = false;
+
+        // TODO: Not sure if we want to add this or if it has an effect etc.
+        // Maybe we want to clear it?
+        // Maybe we'd be able to remember what was the previously selected active element if we don't get rid of it? That might be kinda nice too
+        //this.setActiveElement(undefined);
+
+        const canvas = document.getElementsByClassName(
+            "bubble-edit-generated"
+        )[0];
+        if (canvas && canvas.parentElement) {
+            Comical.convertCanvasToSvgImg(canvas.parentElement as HTMLElement);
+        }
+        Array.from(
+            document.getElementsByClassName("bloom-hideImageButtons")
+        ).forEach(e => e.classList.remove("bloom-hideImageButtons"));
+
+        // Clean up event listeners that we no longer need
+        Array.from(document.getElementsByClassName("bloom-editable")).forEach(
+            element => {
+                element.removeEventListener(
+                    "focus",
+                    TextOverPictureManager.onFocusSetActiveElement
+                );
+            }
+        );
+    }
+
+    public prepareToSavePage(): void {
+        // Review: do we need to call turnOffComicaling, or is that done in time anyway?
+    }
```

Comment:
Seems it IS getting called anyway, so we can get rid of this method and the stuff that calls it.